frontend/src/metabase/query_builder/components/filters/pickers/DatePickerShortcuts.tsx

```diff
@@ -0,0 +1,190 @@
+import moment from "moment";
+import _ from "underscore";
+
+import { ShortcutButton, Separator } from "./DatePickerShortcuts.styled";
+
+import { FieldDimension } from "metabase-lib/lib/Dimension";
+import { Field } from "metabase-types/types/Field";
+import Filter from "metabase-lib/lib/queries/structured/Filter";
+
+function getDateTimeField(field: Field, bucketing?: string) {
+  const dimension = FieldDimension.parseMBQLOrWarn(field);
+  if (dimension) {
+    if (bucketing) {
+      return dimension.withTemporalUnit(bucketing).mbql();
+    } else {
+      return dimension.withoutTemporalBucketing().mbql();
+    }
+  }
+  return field;
+}
+
+type Option = {
+  displayName: string;
+  init: (filter: any) => any[];
+};
+
+const DAY_OPTIONS: Option[] = [
+  {
+    displayName: t`Today`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      "current",
+      "day",
+      { include_current: true },
+    ],
+  },
+  {
+    displayName: t`Yesterday`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -1,
+      "day",
+      { include_current: false },
+    ],
+  },
+  {
+    displayName: t`Last Week`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -1,
+      "week",
+      { include_current: false },
+    ],
+  },
+  {
+    displayName: t`Last 7 Days`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -7,
+      "day",
+      { include_current: false },
+    ],
+  },
+  {
+    displayName: t`Last 30 Days`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -30,
+      "day",
+      { include_current: false },
+    ],
+  },
+];
+
+const MONTH_OPTIONS: Option[] = [
+  {
+    displayName: t`Last Month`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -1,
+      "month",
+      { include_current: false },
+    ],
+  },
+  {
+    displayName: t`Last 3 Months`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -3,
+      "month",
+      { include_current: false },
+    ],
+  },
+  {
+    displayName: t`Last 12 Months`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      -12,
+      "month",
+      { include_current: false },
+    ],
+  },
+];
+
+const MISC_OPTIONS: Option[] = [
+  {
+    displayName: t`Specific dates...`,
+    init: filter => [
+      "=",
+      getDateTimeField(filter[1]),
+      moment().format("YYYY-MM-DD"),
+    ],
+  },
+  {
+    displayName: t`Relative dates...`,
+    init: filter => [
+      "time-interval",
+      getDateTimeField(filter[1]),
+      "current",
+      "none",
+    ],
+  },
+  {
+    displayName: t`Exclude...`,
+    init: filter => ["!=", getDateTimeField(filter[1], "day"), 1],
+  },
+];
+
+type Props = {
+  primaryColor?: string;
+  filter: Filter;
+  onFilterChange: (filter: any[]) => void;
```

Comment:
Can `Filter[]` type be used here instead of any?

Reply:
Done